### ISSUE TITLE
Pull functionality missing for standard abapgit

### ISSUE THREAD

**Issue by marcellourbani** [abapGit/ADT_Backend#39]
Yesterday I started working on it, I have my own proprietary abapgit interface I created at work, which does pull but mix it with lots of unrelated proprietary stuff. So rather than open sourcing that I started working on making this work with this.

The code to do it is commented, so I started an uncomment and fix exercise, but stopped almost immediately: method zcl_abapgit_res_repo_pull->validate_request_data has several issues:

- tries to validate package and url sent by the request, but the eclipse plugin sends none of that (I even debugged it to make it sure, there simply isn't any code to do that)
- validates that a valid transport is sent, even when the package is local
- validates that the package and url don't belong to an existing project, which they must for a pull to make sense

Looks like it was designed to create an abapgit project and then partially changed.
I have no problems fixing that (except I'll be on holidays for the next 10 days) but the change will be quite invasive.
Perhaps was a copy of zcl_abapgit_res_repos->import_repository, which does currently work.

I understand this is more or less a fork of what is actually running on the cloud platform, and would like it to diverge as little as possible. Above all, I don't want to make and maintain my own fork.
Would be great if you guys could sync this with the cloud version as much as possible, or give me some guidance on how things are supposed to work.

**Comment by absap** [comment]
Please consider this repository as a highly experimental. 

The basis of this project is indeed from the version in cloud platform, but that shouldn't be a reason to keep everything as it is. Also, please feel free to make invasive changes. The idea behind this repository is to make it work with the zabapgit report.

You just need to take care of the request versions in the handlers otherwise this could cause problems with the ADT Plugin. So it's my recommendation to edit everything in the HTTP methods (`post`, `get` ...) inside the *`_res_repo`* handlers.

Long story short, just go ahead and feel free to break it :)
